### PR TITLE
=str: TickSource is now a GraphStage

### PR DESCRIPTION
**Pull request by drewhk**
Another custom actor bites the dust

### CODE REVIEW

**Comment by drewhk** [comment]
I had to change the infrastructure slightly to accomodate for materialized values. Currently no materialized value support for fused islands but a temporary solution which works for the single-stage islands we currently have.

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 1]
```diff
@@ -240,4 +245,49 @@ object GraphStages {
     override def toString = "Balance"
   }
 
+  private object TickSource {
+    class TickSourceCancellable(cancelled: AtomicBoolean) extends Cancellable {
+      @volatile var callback: AsyncCallback[Unit] = _
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Ah stupid me, a Promise will work just as good.

**Comment by patriknw** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 2]
```diff
@@ -240,4 +246,53 @@ object GraphStages {
...
+          override def onPull() = () // Do nothing
+        })
+
+        override protected def onTimer(timerKey: Any) =
+          if (isAvailable(out)) push(out, tick)
+      }
+
+      (logic, cancellable)
+    }
+  }
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: `override def toString = "TickSource"`?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: good point, missed that.

**Pull request merged**